### PR TITLE
Loosen restrictions for aggregate propagation to not require aggregator in bitfield

### LINKED ISSUE #958: Require "seen" aggregates to be valid to bound cache
https://github.com/ethereum/eth2.0-specs/pull/1749

### LINKED ISSUE #964: Loosen restrictions for aggregate propagation to not require aggregator in bitfield
https://github.com/ethereum/eth2.0-specs/pull/1801

### PR DESCRIPTION
**Pull request by twoeths**
## Goals
+ resolves #964 
+ resolves #958

### CODE REVIEW

**Comment by twoeths** [comment]
`The valid aggregate attestation defined by `hash_tree_root(aggregate)` has _not_ already been seen (via aggregate gossip, within a verified block, or through the creation of an equivalent aggregate locally).` => not sure how do we check if a aggregated attestation stays in any verified blocks (currently we only check against AggregateAndProofDb - opPool)

File: packages/lodestar/src/network/gossip/validator.ts [hunk 1]
```diff
@@ -175,8 +180,8 @@ export class GossipMessageValidator implements IGossipMessageValidator {
       return false;
     }
 
-    const attestorIndices = getAttestingIndices(this.config, state, attestationData, aggregate.aggregationBits);
-    if (!attestorIndices.includes(aggregatorIndex)) {
+    const committee = getBeaconCommittee(this.config, state, attestationData.slot, attestationData.index);
```
Comment (wemeetagain) [inline packages/lodestar/src/network/gossip/validator.ts]: We should have access to the committee in our `chain.epochCtx` cache.

I bet that we have many places in the gossip validation logic that could be refactored to use chain.epochCtx, might be worth tackling as as separate PR/issue.

**Review by wemeetagain — APPROVED** [review]

File: packages/lodestar/src/network/gossip/validator.ts (hunk 1 above)
Comment (twoeths) [reply packages/lodestar/src/network/gossip/validator.ts]: I only see `getBeaconProposer` and `getBeaconCommittee` in EpochContext that can be applied so I applied that to this PR

File: packages/lodestar/src/sync/gossip/handler.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ export class BeaconGossipHandler implements IGossipHandler {
   };
 
   private onAggregatedAttestation = async (aggregate: SignedAggregateAndProof): Promise<void> => {
-    await this.chain.receiveAttestation(aggregate.message.aggregate);
+    await this.db.aggregateAndProof.add(aggregate.message);
```
Comment (twoeths) [inline packages/lodestar/src/sync/gossip/handler.ts]: need confirmation

Comment (wemeetagain) [reply packages/lodestar/src/sync/gossip/handler.ts]: I think this should still be passed to the chain?
We need to process incoming attestations for the fork choice

Comment (twoeths) [reply packages/lodestar/src/sync/gossip/handler.ts]: I'm not sure, currently we process attestations from received block so I thought this is to put it to the OpPool only to produce block and once the block is published we'll finally process attestations from block.

Comment (twoeths) [reply packages/lodestar/src/sync/gossip/handler.ts]: if we pass to the chain here we may process again when the block containing the same attestation is published, isn't it a duplicate?

Comment (wemeetagain) [reply packages/lodestar/src/sync/gossip/handler.ts]: the fork choice should handle duplicates by only counting the latest one received per attester.

I believe this is where the LMD (latest message driven) aspect of the fork choice shines through. We just feed it all valid messages, and it handles new/duplicate/updated attestations.

Comment (twoeths) [reply packages/lodestar/src/sync/gossip/handler.ts]: I see, thanks for the clarification

File: packages/lodestar/src/chain/chain.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
   public chainId: Uint16;
   public networkId: Uint64;
   public clock: IBeaconClock;
-  private epochCtx: EpochContext;
```
Comment (mpetrunic) [inline packages/lodestar/src/chain/chain.ts:62]: I wonder if it would be better to expose getter to copy of EpochContext as this posses danger that some code might change EpochContext accidentally. 

We could expose getBeaconProposers and getBeaconCommittee from chain which we also need in api.

Comment (wemeetagain) [reply packages/lodestar/src/chain/chain.ts:62]: :+1: definitely anything returned by the API, we can't rely on being unmodified

Comment (wemeetagain) [reply packages/lodestar/src/chain/chain.ts:62]: I think we can do this as a separate PR tho? what do yall think?

Comment (mpetrunic) [reply packages/lodestar/src/chain/chain.ts:62]: I hoped @tuyennhv would do it in separate PR since couple of PR-s depends on this

**Review by wemeetagain — APPROVED** [review]

File: packages/lodestar/src/chain/chain.ts (hunk 3 above)
Comment (twoeths) [reply packages/lodestar/src/chain/chain.ts:62]: sure
